fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -308,6 +303,28 @@ private void checkPartitionColumnWithBaseTable(CreateMaterializedViewStatement s
                 throw new SemanticException("Materialized view related base table partition type:" +
                         partitionInfo.getType().name() + "not supports");
             }
+            replaceTableAlias(slotRef, statement, tableNameTableMap);
+        }
+
+        private SlotRef getSlotRef(Expr expr) {
+            if (expr instanceof SlotRef) {
+                return ((SlotRef) expr);
+            } else {
+                List<SlotRef> slotRefs = Lists.newArrayList();
+                expr.collect(SlotRef.class, slotRefs);
+                Preconditions.checkState(slotRefs.size() == 1);
+                return slotRefs.get(0);
+            }
+        }
+
+        private void replaceTableAlias(SlotRef slotRef,
+                                       CreateMaterializedViewStatement statement,
+                                       Map<TableName, Table> tableNameTableMap) {
+            if (slotRef.getTblNameWithoutAnalyzed().getDb() == null) {
```

Comment:
why should replay table name for this case? please add some comment and cases.

Reply:
Because table may have alias . if TableName replace it with table name , we can use TableName to get table